### ISSUE TITLE
MassaStation won't shutdown

### ISSUE THREAD

**Issue by thomas-senechal** [massalabs/station#740]
**The problem(s)**
After a `CTRL+C`, MassaStation refuses to stop and I have to close it from the Activity Monitor. We saw this bug only happen 3 times within 2 weeks:
1. during an internal test: it happened on a MacBook Pro 13 M1 of a Core team member.
2. during a public presentation: same computer as above
3. right now while testing a PR which did not changed anything about how we handle shut down. Code as [this commit](https://github.com/massalabs/thyra/tree/1a58cf346fc665cc92dc9c3f9e0ea7fe6c99f025)

**To Reproduce**
1. Start MassaStation from a Terminal with `massastation -node-server BUILDNET`
2. Let it run a few seconds
3. `CTRL+C` to shut it down

**Expected behavior(s)**
MassaStation should shut down without any issue

**Screenshots**
<img width="691" alt="image" src="https://github.com/massalabs/thyra/assets/44696378/47d28e15-6ba3-46c3-b04a-2fe78d7bdd27">

**Thyra version**
- 1 and 2 -> `main` at https://github.com/massalabs/thyra/tree/21a4f28df08192453c066bae989c9c67dbc2970f
- 3 -> a PR commit https://github.com/massalabs/thyra/tree/1a58cf346fc665cc92dc9c3f9e0ea7fe6c99f025

**Desktop (please complete the following information):**
 - OS: MacOS
 - Model: MacBook Pro M1 13", MacBook Pro M1 Pro 14"
 
 **Any other useful info**
- During the 3 instances of this bug, the Wallet plugin was installed on v0.0.12

**Comment by Thykof** [comment]
could it be a duplicate of https://github.com/massalabs/station/issues/672? anyway I believe https://github.com/massalabs/station/pull/933 will solve.

**Comment by Thykof** [comment]
I merged #933, could you try again to see if the bug is still here?

**Comment by thomas-senechal** [comment]
Since the bug appears randomly, we can't be sure it is fixed :/
I tried to start/stop MS around 10 times, and it didn't appeared

I guess we can close this issue, if it reappears, we will reopen it (unless a new issue is already opened, in that case we will link it)

**Comment by SlnPons** [comment]
Re-opening the issue as I just had the problem testing the latest version of the main branch. 
However it happens randomly so hard to reproduce. I'll try to deep dive and find a way to reproduce it. Will try to record it.

**Comment by SlnPons** [comment]
Hey @thomas-senechal can you help me to test this issue please? 
As you and I had the bug I would like someone else to check too

**Comment by Thykof** [comment]
I reproduced and tested the fix it works well. 

To reproduce: navigate to the search page and quit Massa Station. 

Before: it's get stuck

after: Massa Station closes after 5 seconds
